### ISSUE TITLE
Error in checkForRemoteErrors(val) : 2 nodes produced errors; first error: NetCDF: DAP failure

### ISSUE THREAD

**Issue by Janelle88** [mjwoods/RNetCDF#93]
I'm running a package ([cft](https://www.earthdatascience.org/cft/index.html)) that relies on some functions from the RNetCDF package, specifically the `read.nc` function and am running into errors. In addition to the error above, I have also gotten the error `Error in try(.Call(R_nc_inq_grp_parent, ncid), silent = TRUE) : NetCDF: No group found.` and `no non-missing arguments to min; returning Infno non-missing arguments to max; returning -Inf` 

I am unsure of why this is happening, but it originally seemed to be in relation to R version 4.1, but I am now getting these errors using R version 4.0.5.

**Comment by mjwoods** [comment]
Hi @Janelle88,

I'm happy to investigate your problem, but can you please provide some more details about the commands you are running or the dataset you are trying to read? Also, which operating system are you running on, and which netcdf library version is installed?

**Comment by Janelle88** [comment]
I just updated RNetCDF to version 2.5.2, and I am running R on Windows 10, R studio version 1.4.1106, R version 4.1.0, but the problem seems to also happen in R 4.0.5. In the cft package, it looks like the code that is being used is:

```
 any_historical <- !is.na(url_pair[1])
    if (any_historical) {
      nc <- RNetCDF::open.nc(url_pair[1])
      historical_array <- RNetCDF::read.nc(nc)[[1]]
      RNetCDF::close.nc(nc)
    } else {
      historical_array <- NA
    }
    
    any_future <- !is.na(url_pair[2])
    if (any_future) {
      nc <- RNetCDF::open.nc(url_pair[2])
      future_array <- RNetCDF::read.nc(nc)[[1]]
      RNetCDF::close.nc(nc)
    } else {
      future_array <- NA
    }
```

or possibly:


```
if (crs == "+proj=longlat +datum=WGS84 +init=epsg:4326" &
                                extent$latmin == 25.0631 & extent$latmax == 49.3960 &
                                extent$lonmin == -124.7722 & extent$lonmax == -67.0648 &
                                resolution == 0.0417){
                              nc <- RNetCDF::open.nc(paste0("https://cida.usgs.gov/",
                                                            "thredds/dodsC/macav2metdata_daily_future",
                                                            "?lat[0:1:584],lon[0:1:1385]"))
                              nc_data <- RNetCDF::read.nc(nc)
                              lats <<- as.vector(nc_data$lat)
                              lons <<- as.vector(nc_data$lon)
                            }
```

I'm not the owner of the package, but this seems to be where they reference RNetCDF. Let me know if there's any more information you need.

**Comment by mjwoods** [comment]
Thanks for the extra information, but I still need a way to reproduce the problem. When you run the package, what sequence of commands will produce the error messages shown earlier?

**Comment by Janelle88** [comment]
Ah, yes, this is the function I'm running.

```
library(tidyverse)
library(here)
library(cft)

latitude = 35.75758546,
longitude = -106.3054344

directory <- here::here()

Site_coordinates <- data.frame(PARK=SiteID,lat=latitude,lon=longitude)
  sp::coordinates(Site_coordinates) <- ~lon+lat
  sp::proj4string(Site_coordinates) <- "+proj=longlat +datum=NAD83 +no_defs "

  file_refs <- cft::cftdata(aoi = Site_coordinates,
                            area_name = "BAND",
                            years = c(1950, 2099),
                            models = c('bcc-csm1-1','bcc-csm1-1-m','BNU-ESM','CanESM2','CCSM4','CNRM-CM5','CSIRO-Mk3-6-0',
                            'GFDL-ESM2G','GFDL-ESM2M','HadGEM2-CC365','HadGEM2-ES365',  'inmcm4','IPSL-CM5A-MR',
                            'IPSL-CM5A-LR','IPSL-CM5B-LR','MIROC5','MIROC-ESM','MIROC-ESM-CHEM','MRI-Cgcm3','NorESM1-M'),
                            local_dir = directory,
                            parameters = c("pr", "tasmax", "tasmin", "rhmin", "rhmax"),
                            scenarios =  c("rcp45", "rcp85"),
                            ncores = parallel::detectCores() / 2)
```

**Comment by mjwoods** [comment]
Hi again.

Strangely, those commands worked for me on Windows 10 in R-4.1.1 (64-bit). I tested both RNetCDF versions 2.4-2 and 2.5-2.

I did reduce the range of years from 1950-2099 to 2000-2002, to reduce the amount of data downloaded over my internet connection.

I also ran with `ncores=4`, so maybe the number of cores has something to do with your problem. 

Maybe one of the remote sites was down when you tried it? Your network could block some remote sites too - for "security reasons".

Regards,
Milton.
